### PR TITLE
Endpoint added for the registration of a user

### PR DESCRIPTION
**Pull request by subodh101**
- POST request for user registration
- Added a new structure for the list of users.

To Do: Add the registering user into the database.

### CODE REVIEW

File: data/data.go
Comment (BharathTalakad) [inline data/data.go:20]: I believe that there is no need to create a list of users, since we will never fetch some users at anytime. 
After Unmarshalling the json it is sufficient to make a call to a function that handles the writes to the database by passing just the single user json.

File: models/models.go
Comment (BharathTalakad) [inline models/models.go:87]: There is no need for this structure

**Review by BharathTalakad — CHANGES_REQUESTED** [review]

File: server/handlers.go
Comment (BharathTalakad) [inline server/handlers.go:17]: There is a better approach involving the use of gin context itself
`c.BindJSON(&user)`
This automatically binds the input json to the `var user` struct variable

File: data/data.go
Comment (subodh101) [reply data/data.go:20]: Yes, You are right! Let's hold this PR till tomorrow. Once the DB is ready will make those specific changes

File: models/models.go
Comment (subodh101) [reply models/models.go:87]: Yes, then this will also be removed.

File: server/server_test.go
Comment (BharathTalakad) [inline server/server_test.go:64]: A better way to test would be call the function to read from database. And check if the unmarshalled data from database same as json data passed to the POST userRegistration

**Review by BharathTalakad — CHANGES_REQUESTED** [review]
I will wait for how team-cyan writes the database calls before merging this pull request

File: server/handlers.go
Comment (subodh101) [reply server/handlers.go:17]: Cool! thanks!

File: server/server_test.go
Comment (subodh101) [reply server/server_test.go:64]: yes! that makes sense!

**Review by BharathTalakad — APPROVED** [review]

**Pull request closed without merging**
